scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1342,10 +1344,8 @@ private void sendMessage(final RawData message, final DTLSSession session) {
 					session.getSequenceNumber(),
 					new ApplicationMessage(message.getBytes(), message.getInetSocketAddress()),
 					session);
-			if (message.getMessageCallback() != null) {
-				message.getMessageCallback().onContextEstablished(ctx);
-			}
```

Comment:
After my answer to your first comment about the order I was already starting to think in the same direction as your second comment :-).
`MassageCallback` requires 3 methods, stay with `onContextEstablished()` and the 2 additional ones `onSent()` and `onSendError()`. I would stay with calling `onContextEstablished()` before sending and `onSent()` afterwards or `onSendError()`, if sending fails.